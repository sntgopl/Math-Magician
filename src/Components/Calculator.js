import React from 'react';
import ReactDOM from 'react-dom/client';

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      active: false,
    }
  }

  render() {
    return (
      
      <div className="Calculator">
        <p className="result">0</p>
        <div className="board">
          <button type="button" onClick={}>AC</button>
          <button type="button" onClick={}>+/-</button>
          <button type="button" onClick={}>%</button>
          <button type="button" onClick={}>/</button>
          <button type="button" onClick={}>7</button>
          <button type="button" onClick={}>8</button>
          <button type="button" onClick={}>9</button>
          <button type="button" onClick={}>X</button>
          <button type="button" onClick={}>4</button>
          <button type="button" onClick={}>5</button>
          <button type="button" onClick={}>6</button>
          <button type="button" onClick={}>-</button>
          <button type="button" onClick={}>1</button>
          <button type="button" onClick={}>2</button>
          <button type="button" onClick={}>3</button>
          <button type="button" onClick={}>+</button>
          <button type="button" onClick={}>0</button>
          <button type="button" onClick={}>.</button>
          <button type="button" onClick={}>=</button>
        </div>
      </div>
    )
  }
}

