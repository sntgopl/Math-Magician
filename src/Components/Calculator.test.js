import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import { screen } from '@testing-library/dom';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import Calculator from './Calculator';

it('renders correctly', () => {
  const tree = renderer
    .create(<Calculator />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe('Test If a user click a button on the Calculator  the display should Update', () => {
  it('If a user click AC button the Calculator Screen is Blank', () => {
    render(<Calculator />);
    userEvent.click(screen.getByRole('button', { name: /ac/i }));

    expect(screen.getByTestId('display')).toBeInTheDocument();
  });

  it('If a user click 7 and 8 button the Calculator Screen displays 78', () => {
    render(<Calculator />);
    userEvent.click(screen.getByRole('button', { name: /7/i }));
    userEvent.click(screen.getByRole('button', { name: /8/i }));
    expect(screen.getByText(/78/i)).toBeInTheDocument();
  });

  it('If a user click 63, +, and 8 button the Calculator Screen displays 63 + 8', () => {
    render(<Calculator />);
    userEvent.click(screen.getByRole('button', { name: /6/i }));
    userEvent.click(screen.getByRole('button', { name: /3/i }));
    userEvent.click(screen.getByRole('button', { name: /x/i }));
    userEvent.click(screen.getByRole('button', { name: /8/i }));
    expect(screen.getByText(/63\x8/i)).toBeInTheDocument();
  });

  it('If a user click 63, +, and 8 button the Calculator Screen displays 63 + 8', () => {
    render(<Calculator />);
    userEvent.click(screen.getByRole('button', { name: /6/i }));
    userEvent.click(screen.getByRole('button', { name: /x/i }));
    userEvent.click(screen.getByRole('button', { name: /8/i }));
    userEvent.click(screen.getByRole('button', { name: /=/i }));
    expect(screen.getByText(/48/i)).toBeInTheDocument();
  });
  it('If a user click 6, +, 8 and then = button the Calculator Screen displays 14', () => {
    render(<Calculator />);
    userEvent.click(screen.getByRole('button', { name: /6/i }));
    userEvent.click(screen.getByRole('button', { name: /8/i }));
    expect(screen.getByText(/68/i)).toBeInTheDocument();
  });
});
