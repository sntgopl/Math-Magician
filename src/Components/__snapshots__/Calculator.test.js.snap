// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Calculator"
>
  <p
    className="result"
    data-testid="display"
  >
     
    0
  </p>
  <div
    className="board"
  >
    <button
      data-testid="AC"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="btn0"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
